test/js/samples/debug-empty/expected.js

```diff
@@ -19,18 +19,16 @@ const file = undefined;
 
 function create_fragment(ctx) {
 	let h1;
+	let t0_fn = ctx => `Hello ${ctx.name}!`;
+	let t0_value = t0_fn(ctx);
```

Comment:
we should be able to hoist these functions out of the block altogether, I think?

(I still have an ambition to merge the create/update phases so that we can just 'update' everything to its initial state, rather than having this duplication at all. But that's a future nice-to-have)